typings/index.d.ts

```diff
@@ -293,6 +296,9 @@ export interface Thenable {
 export interface RenderOptions {
```

Comment:
I check RTL/DTL and they seem to have quite decent TS typings for custom queries:

```typescript
export type Query = (
    container: HTMLElement,
    ...args: any[]
) => Error | Promise<HTMLElement[]> | Promise<HTMLElement> | HTMLElement[] | HTMLElement | null;

export interface Queries {
    [T: string]: Query;
}
```
Source: https://github.com/testing-library/dom-testing-library/blob/master/types/get-queries-for-element.d.ts

```typescript
export type RenderResult<Q extends Queries = typeof queries> = {
  // ...
} & {[P in keyof Q]: BoundFunction<Q[P]>}

export interface RenderOptions<Q extends Queries = typeof queries> {
  // ...
  queries?: Q
}
```
Source: https://github.com/testing-library/react-testing-library/blob/master/types/index.d.ts

IMO we could have the same at least for TS, possibly for flow as well if possible.

Reply:
Please note that the `Query` type restricts return value to (Promise)(Array) Element + null + Error. IMO if we provide element queries then we should also consider similar restriction. wdyt?